pkg/clientaccess/clientaccess.go

```diff
@@ -73,138 +74,122 @@ func WriteClientKubeConfig(destFile, url, serverCAFile, clientCertFile, clientKe
 }
 
 type Info struct {
-	URL      string `json:"url,omitempty"`
 	CACerts  []byte `json:"cacerts,omitempty"`
-	username string
-	password string
-	Token    string `json:"token,omitempty"`
-}
-
-func (i *Info) ToToken() string {
-	return fmt.Sprintf("K10%s::%s:%s", hashCA(i.CACerts), i.username, i.password)
+	BaseURL  string `json:"baseurl,omitempty"`
+	Username string `json:"username,omitempty"`
+	Password string `json:"password,omitempty"`
+	caHash   string
 }
 
-func NormalizeAndValidateTokenForUser(server, token, user string) (string, error) {
-	if !strings.HasPrefix(token, "K10") {
-		token = "K10::" + user + ":" + token
-	}
-	info, err := ParseAndValidateToken(server, token)
-	if err != nil {
-		return "", err
-	}
-
-	if info.username != user {
-		info.username = user
-	}
-
-	return info.ToToken(), nil
+// String returns the token data, templated according to the token format
+func (info *Info) String() string {
+	return fmt.Sprintf(tokenFormat, tokenPrefix, hashCA(info.CACerts), info.Username, info.Password)
```

Comment:
I have a question here, does this work with --token <secret> for example, I think the check was put here to handle where the user specify a custom token.

Reply:
Yes, it works regardless of whether the user provided a short token (just the password) or a full token including the CA hash and username. This is basically the same thing as what the ToToken function previously did, and is used in a few places to normalize the token string.